src/gui/dialogs/SpellcheckLanguageDialog.js

```diff
@@ -48,18 +48,18 @@ async function getItems(): Promise<Array<SelectorItem<string>>> {
 		...options.map(code => {
 
 			const [langCode, locale] = code.split("-")
+			// first, find the name for a language given a locale with a perfect match
+			const l = languages.find(language => (locale && language.code === `${langCode}_${locale.toLowerCase()}`))
+				// find the name for a language without a locale, with a perfect match
+				|| languages.find(language => language.code === langCode)
+				// try to get a missing one before splitting
+				|| getMissingLanguage(langCode)
+				// the code given by electron doesn't always have a locale when we do,
+				// e.g. for Persian we have "fa_ir" in LanguageViewModel, but electron only gives us "fa"
+				|| languages.find(language => language.code.slice(0, 2) === langCode)
+			const textId = mapNullable(l, language => language.textId)
```

Comment:
`mapNullable` has been made redundant by optional chaining, we should use that instead: `textId = l?.textId`. Also `l` should be named `language`